oak_crypto/src/tests.rs

```diff
@@ -0,0 +1,211 @@
+//
+// Copyright 2023 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use crate::{
+    aead::{Aead, AEAD_ALGORITHM_KEY_SIZE_BYTES, AEAD_NONCE_SIZE_BYTES},
+    hpke::Hpke,
+    kem::Kem,
+    util::i2osp,
+    ClientCryptoProvider, EnclaveCryptoProvider, MessageWithAssociatedData,
+};
+
+/// Test AES-GCM key that is only used in tests.
+/// Was generated by calling [`Hpke::setup_base_sender`].
+const TEST_AEAD_KEY: [u8; AEAD_ALGORITHM_KEY_SIZE_BYTES] = [
+    11, 107, 5, 176, 4, 145, 171, 193, 163, 81, 105, 238, 171, 115, 56, 160, 130, 85, 22, 227, 118,
+    76, 77, 89, 144, 223, 10, 112, 11, 149, 205, 199,
+];
+/// Test AES-GCM nonce that is only used in tests.
+const TEST_NONCE: [u8; AEAD_NONCE_SIZE_BYTES] = [
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
```

Comment:
Worth using a different (non-zero) nonce, to at least have some reassurance it is not being accidentally ignored

Reply:
Done